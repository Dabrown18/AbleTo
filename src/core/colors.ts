const colors = {
    white: '#fff',
    black: '#000',
    creamBackground: '#F7F5EF',

    gray50: '#F8F8F8',
    gray100: '#EFEFEF',
    gray200: '#E1E1E1',
    gray300: '#D8D7D7',
    gray400: '#C9C7C5',
    gray500: '#A6A29E',
    gray600: '#787471',
    gray700: '#5E5C59',
    gray800: '#4B4845',
    gray900: '#2D2926',

    primary50: '#F3E8F0',
    primary100: '#E5CADE',
    primary200: '#D49FCA',
    primary300: '#C177B6',
    primary400: '#A94E99',
    primary500: '#913E84',
    primary600: '#85356F',
    primary700: '#662955',
    primary800: '#491D3D',
    primary900: '#2C1125',

    orange50: '#FDF0ED',
    orange100: '#F8D3C9',
    orange200: '#F3B6A5',
    orange300: '#EB896E',
    orange400: '#E86C4A',
    orange500: '#E35129',
    orange600: '#C73F1A',
    orange700: '#902E13',
    orange800: '#591C0C',
    orange900: '#360F07',

    cream50: '#F1EFE5',
    cream100: '#E5E3D8',
    cream200: '#D4D2C8',
    cream300: '#BBB9B0',
    cream400: '#A3A098',
    cream500: '#88867F',
    cream600: '#716F6A',
    cream700: '#595753',
    cream800: '#41403D',
    cream900: '#221F1D',

    blue50: '#EBF1F9',
    blue100: '#D6E4FA',
    blue200: '#B6D1ED',
    blue300: '#8FBDE6',
    blue400: '#78AAE6',
    blue500: '#588ECC',
    blue600: '#4678B4',
    blue700: '#2D598A',
    blue800: '#1C416A',
    blue900: '#0D2B49',

    yellow50: '#FEF8EC',
    yellow100: '#FBEBC6',
    yellow200: '#F8DEA0',
    yellow300: '#F3C353',
    yellow400: '#E5A510',
    yellow500: '#BF890D',
    yellow600: '#996E0B',
    yellow700: '#725208',
    yellow800: '#4C3705',
    yellow900: '#3d2412',

    teal50: '#E7F3F3',
    teal100: '#D0E7E7',
    teal200: '#B2D9D9',
    teal300: '#96C8C8',
    teal400: '#45AB9C',
    teal500: '#30948D',
    teal600: '#2B7D82',
    teal700: '#275F63',
    teal800: '#234548',
    teal900: '#192D2F',

    red50: '#FDEDED',
    red100: '#FADBDB',
    red200: '#F8C5C5',
    red300: '#F5A3A3',
    red400: '#F27E7E',
    red500: '#F04949',
    red600: '#D42D2D',
    red700: '#A82121',
    red800: '#7E1616',
    red900: '#570C0C',

    focusShadow: '#588ECC',
    blackTransparent: 'rgba(26, 33, 45, 0.6)',
    grayTransparent700: '#5e5c5966',
    creamTransparent200: '#D4D2C84D',
    primaryTransparent600: '#85356FCC',
    grayTransparent900: '#2D292680',
    primaryTransparent50: '#F3E8F000',
} as const;

export type Color = keyof typeof colors;

export default colors;
