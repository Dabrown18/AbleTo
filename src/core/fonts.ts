const fonts = {
    FAKT_PRO_NORMAL: 'fakt_pro_normal',
    FAKT_PRO_MEDIUM: 'fakt_pro_medium',
    GUARDIAN_EGYP_LIGHT: 'guardian_egyp_light',
};

export default fonts;
