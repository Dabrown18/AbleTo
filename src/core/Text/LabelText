import React, { FunctionComponent } from 'react';
import { Text, TextProps } from 'react-native';
import styled from 'styled-components/native';

import colors from '../colors';
import fonts from '../fonts';
import InterpolatedText from './interpolation/InterpolatedText';

export type LabelVariant = 'l1' | 'l2';

type Props = {
    variant?: LabelVariant;
} & TextProps;

const LabelText: FunctionComponent<Props> = ({ variant = 'l1', ...rest }) => {
    return <InterpolatedText component={StyledText} componentProps={{ variant, ...rest }} {...rest} />;
};

const StyledText = styled(Text)<{ variant: LabelVariant }>`
    font-family: ${(props) => styles[props.variant].fontFamily};
    color: ${(props) => styles[props.variant].color};
    font-weight: ${(props) => styles[props.variant].fontWeight};
    font-size: ${(props) => styles[props.variant].fontSize};
    line-height: ${(props) => styles[props.variant].lineHeight};
    letter-spacing: ${(props) => styles[props.variant].letterSpacing};
`;

const styles = {
    l1: {
        fontFamily: fonts.FAKT_PRO_MEDIUM,
        color: colors.gray800,
        fontWeight: '500',
        fontSize: '12px',
        lineHeight: '16px',
        letterSpacing: '0.4px',
    },
    l2: {
        fontFamily: fonts.FAKT_PRO_MEDIUM,
        color: colors.gray800,
        fontWeight: '500',
        fontSize: '14px',
        lineHeight: '21px',
        letterSpacing: '0.5px',
    },
};

export default LabelText;
