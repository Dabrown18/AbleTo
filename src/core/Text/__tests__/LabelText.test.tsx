import { render } from '@testing-library/react-native';
import React from 'react';
import { ThemeProvider } from 'styled-components/native';

import LabelText, { LabelVariant } from '../LabelText';
import colors from '@src/core/colors';
import fonts from '@src/core/fonts';

describe('BodyText snapshots', () => {
    const theme = {
        colors,
        fonts,
    };

    const renderText = (variant: LabelVariant) =>
        render(
            <ThemeProvider theme={theme}>
                <LabelText variant={variant} children="Hello there!" />
            </ThemeProvider>
        );

    test('Label 1', () => {
        const tree = renderText('l1');

        expect(tree).toMatchSnapshot();
    });

    test('Label 2', () => {
        const tree = renderText('l2');

        expect(tree).toMatchSnapshot();
    });
});
