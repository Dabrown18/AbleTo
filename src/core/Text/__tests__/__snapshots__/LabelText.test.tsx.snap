// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyText snapshots Label 1 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4B4845",
        "fontFamily": "fakt_pro_medium",
        "fontSize": 12,
        "fontWeight": "500",
        "letterSpacing": 0.4,
        "lineHeight": 16,
      },
    ]
  }
  variant="l1"
>
  Hello there!
</Text>
`;

exports[`BodyText snapshots Label 2 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4B4845",
        "fontFamily": "fakt_pro_medium",
        "fontSize": 14,
        "fontWeight": "500",
        "letterSpacing": 0.5,
        "lineHeight": 21,
      },
    ]
  }
  variant="l2"
>
  Hello there!
</Text>
`;
